### PR TITLE
Addding AbstractCloudBigtableTableDoFn for complex user CBT DoFns.

### PR DESCRIPTION
**Pull request by sduskis**
Multiple users contacted us about DoFns that require some additional Gets, including issue #607. This PR should help.

### CODE REVIEW

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import org.slf4j.LoggerFactory;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+
+/**
+ * This class can be used as a superclass for {@link DoFn}s that require a {@link Connection} in
+ * Dataflow.
+ * 
+ * @param <In> the type of the (main) input elements
+ * @param <Out>  the type of the (main) output elements
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java]: nit: remove an extra white space between "<Out>" and "the"

File: bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ *
```
Comment (kevinsi4508) [inline bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java]: remove an extra new line

**Comment by kevinsi4508** [comment]
LGTM

**Comment by sduskis** [comment]
I updated the javadoc.  Good catch.

**Pull request merged**